settings/settings.go

```diff
@@ -88,19 +106,37 @@ func (cfg *Config) LoadFromDisk() error {
 		return err
 	}
 
-	err = yaml.Unmarshal(content, &cfg)
+	var cfgYAML configYAML
+	err = yaml.Unmarshal(content, &cfgYAML)
+
+	var cfgKeyring configKeyring
+	if token, err := keyring.Get("circleci", "circleci_cli"); err == nil {
+		cfgKeyring.Token = token
+	}
+
+	cfg.merge(&cfgYAML, &cfgKeyring)
```

Comment:
We should fall back to using the configuration, rather than trying to exclusively use the keychain. This covers edge cases where it isn't available.